Decred Wallet/Extensions/UIButton.swift

```diff
@@ -54,4 +53,22 @@ extension UIButton {
             self.setNeedsLayout()
         }
     }
+
+    private func imageWithColor(color: UIColor) -> UIImage? {
+        let rect = CGRect(x: 0.0, y: 0.0, width: 1.0, height: 1.0)
+        UIGraphicsBeginImageContext(rect.size)
+        let context = UIGraphicsGetCurrentContext()
+        
+        context?.setFillColor(color.cgColor)
+        context?.fill(rect)
+        
+        let image = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        
+        return image
+    }
+    
+    func setBackgroundColor(_ color: UIColor, for state: UIControl.State) {
```

Comment:
The following approach is simpler. I don't see a need to create another function that is then called from this function when that other function is not needed else where nor is this function already complicated and needs to be simplified.

Let me rephrase. Here are 2 good reasons for creating functions:
- If a particular section of code becomes complex, you create new function(s) to hold some of the complex codes and invoke those functions from the previously complex code. This `setBackgroundColor` function is not complex, don't see a need to simplify it by creating a separate function and invoking it from here.

- Another rationale for creating functions is to abstract repeated code. If a block of code appears in multiple places, it becomes wise to refactor that block into a function and invoke that function from the multiple places the block of code appears. The result is that the new function created is used/called from multiple places in code. The `imageWithColor` function isn't called from anywhere else.

Conclusion: The content of the `imageWithColor` function can be placed in this `setBackgroundColor` function directly, no need for that other function.

See this sample code [from SO](https://stackoverflow.com/a/30604658/6086027);
```
func setBackgroundColor(color: UIColor, forState: UIControl.State) {
        self.clipsToBounds = true  // add this to maintain corner radius
        UIGraphicsBeginImageContext(CGSize(width: 1, height: 1))
        if let context = UIGraphicsGetCurrentContext() {
            context.setFillColor(color.cgColor)
            context.fill(CGRect(x: 0, y: 0, width: 1, height: 1))
            let colorImage = UIGraphicsGetImageFromCurrentImageContext()
            UIGraphicsEndImageContext()
            self.setBackgroundImage(colorImage, for: forState)
        }
    }
```